src/data/markdown/translated-guides/en/02 Using k6/06 Test life cycle.md

```diff
@@ -41,33 +62,41 @@ export default function () {
 
 </CodeGroup>
 
-_"Why not just run my script normally, from top to bottom"_, you might ask - the answer is: we
-do, but code **inside** and **outside** your `default` function can do different things.
+Why not just run my script normally, from top to bottom?
+
+This is a good question, and the answer is: we do.
+But code *inside* and *outside* your `default` function do different things.
+
+Code inside `default` is called *VU code*.
+VU code runs over and over through the test duration.
+Besides setup and teardown, code outside of `default` is  *init code*.
+It runs only once per VU.
+
+VU code can make HTTP requests, emit metrics, and generally do everything you'd expect a load test to do.
+There are a few important exceptions. VU code:
+* *Does not* load files from your local filesystem,
+* *Does not* import any other modules. 
```

Comment:
I think that this explains *what* you can do and where, but not *why* which arguably is what the question above asks. 

The short answer is that "we need the script to be executed in order to know what k6 should do". That includes:
- getting hte configuration
- knowing which functions to be called on `setup`/`teardown` and iterations
- all the things above that you have mentioned above are also separated in order to 
  - on one side not generate metrics whenever we have to run the "init code" and to be able to do it fairly repeatable
  - getting an `archive` with everything needed to run the code somewhere else (like in the cloud)  

And that last two points is where a lot of the restrictions come from. But ultimately we just *use* the fact that we need to be able to run "init code" to get the `options` and the definitions to do that.

Also, ultimately running 
```javascript
let b = 5;
```
twice in the same javascript context will give you an error as `b` has already been defined. Putting it in the function and calling *the function* is what makes a new context and let you do 

```javascript
function f () {
  let res = http.request()
  // something with res
}
f();
f();
```
So there is also kind of another technical reason for this.

Reply:
Thanks for these explanations! I think you are getting at something very essential to keep in mind when explaining these internals.

 We always must ask: _what does this information mean to the user? What bad thing will happen if they don't know this information?_

 It's tricky sometimes to separate what information matters for implementers from what matters for end users.